### DISCUSSION TITLE
Feature Request: Add node name to all entity IDs

### DISCUSSION THREAD

**Discussion by EpicLPer** [dougiteixeira/proxmoxve#283] [Feature request] [answered]
Heya,

I've set up 2 Proxmox hosts via this integration and noticed that some entity IDs generated the same name twice, the second node's IDs having a `_2` at the end.  
This happens due to proxmoxve using the same sensor name for both of my nodes, which is as example `sensor.storage_local_disk_used_percentage`. The second node is being set up with the exact same entity ID name, which results in a conflict and home assistant assigning it a `_2` at the end then.

**Comment by dougiteixeira** [comment]
Please open an issue and attach the integration diagnostics file so that I can better identify the problem and how to fix it.

**Reply by EpicLPer** [reply]
Hi, sorry for the late reply. I've since then just manually given the entities their respective names.
I'm not sure if a diagnostic file would have helped in this case either way, the integration creates similar entity names without the hostname of the node so it conflicts with other entity names and generates a `_2` entity.

**Comment by michaelblight** [comment]
I came here with the same request. It would be nice if the configuration had an optional prefix to go at the front of each `entity_id` created. I have two Proxmox instances - 'erying' and 'nuc', and in my case the integration created the following two entities:
```
sensor.storage_local_lvm_disk_used_percentage
sensor.storage_local_lvm_disk_used_percentage_2
```
So the suggestion would be that by entering a prefix during config, I could have them instead named as:
```
sensor.erying_storage_local_lvm_disk_used_percentage
sensor.nuc_storage_local_lvm_disk_used_percentage
```
Obviously this could be done manually, but there's a lot of entities created. Also, if you didn't enter a prefix and put one in later (or changed it), the old entities would need to be deleted.

**Comment by dougiteixeira** [comment, marked as answer]
Included in #304. Available in the next version (3.4.2).

**Comment by mZ738** [comment]
5.2.0 adds an **entity id scheme** choice at setup: *Standard* is Home Assistant's device-then-name as before; *Extended* puts a common prefix first and the id before the name — `sensor.pve_qemu_108_win11_cpu_used`, `sensor.pve_node_pve_cpu_used` — so a search or recorder filter catches everything by the prefix and a list sorts by id. The prefix is `pve` unless you type another. Shared storage carries its node in the id as well, so the `_2` collisions of two nodes' `storage_local_…` are gone. Entities that exist keep their ids — the scheme applies to entities registered from then on, and there is deliberately no bulk rename. https://github.com/dougiteixeira/proxmoxve/releases/tag/5.2.0
